api/src/routes/edit.ts

```diff
@@ -1,10 +1,59 @@
 import express = require("express");
+import bodyParser = require("body-parser");
 import Config from "../models/Config";
-const router = express.Router();
+import Fedora from "../services/Fedora";
+import FedoraObjectFactory from "../services/FedoraObjectFactory";
+import MetadataExtractor from "../services/MetadataExtractor";
 import { requireToken } from "./auth";
 
+const router = express.Router();
+
 router.get("/models", requireToken, function (req, res) {
     res.json({ CollectionModels: Config.getInstance().collectionModels, DataModels: Config.getInstance().dataModels });
 });
 
+router.post("/object/new", requireToken, bodyParser.json(), async function (req, res) {
+    let parentPid = req?.body?.parent;
+    if (parentPid !== null && parentPid.length === 0) {
+        parentPid = null;
+    }
+    const noParent = (req.body.noParent ?? "0") === "1";
```

Comment:
Why do you need noParent and parent parameter? Doesn't the existence of parent parameter mean there is a parent?

Reply:
You make a good point. :-) My thinking here was that the "no parent" case is almost never needed, and accidentally creating an object with no parent could be a headache to fix, so I wanted to make it an opt-in feature. But that is quite possibly over-engineering that can be simplified.

Reply:
I'll give it some more thought next week; unfortunately I don't think I'm going to get much coding time today due to meetings, etc.

Reply:
Yes, on closer inspection, you are absolutely right that this really serves no useful purpose. I have removed it.